NDTensors/test/NDTensorsTestUtils/device_list.jl

```diff
@@ -1,11 +1,14 @@
 using NDTensors: NDTensors
+using Pkg: Pkg
 if "cuda" in ARGS || "all" in ARGS
   using CUDA
 end
 if "rocm" in ARGS || "all" in ARGS
+  Pkg.add("AMDGPU")
```

Comment:
Let's add a comment here and down below to explain that we do this instead of putting it into `Project.toml` because it can't be installed on Julia 1.6.

Reply:
Oh I see, it is in `[deps]`. I guess that's a bit strange but ok.

Reply:
Its in [extra] but I think its a good idea to put a comment about how Metal and AMDGPU don't work on versions lower than 1.x (I have to verify which version)

Reply:
Perhaps we should add back the `[extras]` section. Its not actually used in `Pkg.test` so it should be no problem

Reply:
Oops my second comment was meant for the other comment thread.

I'd be ok adding back `[extras]` and listing these optional test dependencies there. I was never clear on what the `[extras]` section was for, but it could be a good way to keep track of which packages are optional test dependencies in one place.

Reply:
We could name the section `[extensions]` if that makes more sense? Regardless of the name its ignored by `Pkg.test` which only looks at `[deps]`

Reply:
Sounds good.

Reply:
I updated this to `[weakdeps]`